deployment/src/main/java/io/quarkiverse/asyncapi/generator/AsyncApiCodeGenerator.java

```diff
@@ -73,12 +67,33 @@ private String escape(String raw) {
         return escaped;
     }
 
-    private void writeTemplate(String className, String templateName, Map<String, Object> map) throws IOException {
-        Files.writeString(outPath.resolve(Path.of(className + JAVA_SUFFIX)),
+    private String writeTemplate(String id, String templateName, Map<String, Object> map) throws IOException {
+        String simpleClassName = id + templateName;
+        Files.writeString(
+                Files.createDirectories(outPath.resolve(basePackage.replace('.', '/')))
+                        .resolve(simpleClassName + JAVA_SUFFIX),
                 QuteTemplateHelper.getTemplate(config, templateName).data(map).render());
+        return simpleClassName;
     }
 
     public boolean done() throws IOException {
-        return generated;
+        if (!configClassNames.isEmpty()) {
+            writeServiceLoader();
+            return true;
+        }
+        return false;
     }
+
+    private void writeServiceLoader() throws IOException {
+        Path serviceLoader = Files
```

Comment:
Till we have generate resources support in quarkus, we need this trick to directly write the service loader to classpath
This code is making two assumptions:
1) The there are always two dirs from the dir where the code is generated till the root directory of the project
2) That classes is the output dir
These two patsh cannot be changed right now so it will work, but we need to change that in the future

Reply:
What do you think of adding this as a comment?

Reply:
too large ;)